packages/hooks/src/use-pagination-state/use-pagination-state.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useCallback } from 'react';
+
+const defaultValues = {
+  page: 1,
+  perPage: 20,
+};
+
+const usePaginationState = (initialValues = defaultValues) => {
+  const mergedValues = {
+    ...defaultValues,
+    ...initialValues,
+  };
+
+  const [page, setPage] = useState(mergedValues.page);
+  const [perPage, setPerPage] = useState(mergedValues.perPage);
+
+  const onPageChange = useCallback(
+    (nextPage) => {
+      setPage(nextPage);
+    },
+    [setPage]
+  );
+
+  const onPerPageChange = useCallback(
+    (nextPerPage) => {
+      setPage(1);
```

Comment:
Add a comment why this is needed.

Reply:
...and add a test for this case. At least I can't spot one :)

Reply:
good call. I got this.